lib/Extension/LanguageServer/LanguageServerExtension.php

```diff
@@ -506,9 +509,24 @@ private function registerDiagnostics(ContainerBuilder $container): void
                 ...$this->taggedServices($container, self::TAG_CODE_ACTION_DIAGNOSTICS_PROVIDER)
             );
         }, [
-            self::TAG_DIAGNOSTICS_PROVIDER => [
-                'name' => 'code-action'
-            ],
+            self::TAG_DIAGNOSTICS_PROVIDER => DiagnosticProviderTag::create('code-action', outsource: true),
+        ]);
+
+        $container->register(OutsourcedDiagnosticsProvider::class, function (Container $container) {
+            /** @var PathResolver $resolver */
+            $resolver = $container->get(FilePathResolverExtension::SERVICE_FILE_PATH_RESOLVER);
+            $projectPath = $resolver->resolve('%project_root%');
+            // only register this if we should call out to an external process for diagnostics
+            if (!$container->getParameter(self::PARAM_DIAGNOSTIC_OUTSOURCE)) {
+                return null;
+            }
+
+            return new OutsourcedDiagnosticsProvider([
+                __DIR__ . '/../../../bin/phpactor',
```

Comment:
this should be fine, could also use the file path resolver to get the application root